src/arrayMethodReduce.js

```diff
@@ -4,8 +4,15 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(callback, initialValue = this[0]) {
+    let redusedValue = initialValue;
+    if (typeof initialValue === 'string') {
```

Comment:
are you sure you need this?

Reply:
Yes, without this condition, the string test ('takes first item as a `startValue` if not given, strings') fails. I had result:  "aabcd"